### PR TITLE
OpenAL cleanup

### PR DESCRIPTION
**Pull request by devvoid**
While I wait for #908 to be finished so I can put the last touches on #867, I'm going to start on a project I've been meaning to get to for a while.

This is going to be a pretty big cleanup of the OpenAL bindings; as of right now, the compiler generates a lot of warnings, and Rider has a bunch of issues with the code style. Once this is over, it should build without any complaints.

### CODE REVIEW

**Comment by devvoid** [comment]
This is ready for review!

File: src/OpenAL/Extensions/OpenTK.OpenAL.Extensions.Creative.EFX/Presets/ReverbPresets.cs [hunk 1]
```diff
@@ -19,8 +19,7 @@ public static class ReverbPresets
         /// <summary>
         /// A reverb preset (approximating a generic location).
         /// </summary>
-        public static readonly ReverbProperties Generic = new ReverbProperties
-        (
+        public static readonly ReverbProperties Generic = new ReverbProperties(
```
Comment (Perksey) [inline src/OpenAL/Extensions/OpenTK.OpenAL.Extensions.Creative.EFX/Presets/ReverbPresets.cs]:
```cs
LongMethodInvocations
(
    arg,
    anotherArg
)
```
This is how the code should be and is a rule we sadly can't configure in Rider/ReSharper. Revert.

File: src/OpenAL/Extensions/OpenTK.OpenAL.Extensions.Creative.EFX/Presets/ReverbPresets.cs [hunk 2]
```diff
@@ -43,14 +42,12 @@ public static class ReverbPresets
...
-            0x1
-        );
+            0x1);
 
         /// <summary>
         /// A reverb preset (approximating a padded cell).
         /// </summary>
-        public static readonly ReverbProperties PaddedCell = new ReverbProperties
-        (
+        public static readonly ReverbProperties PaddedCell = new ReverbProperties(
```
Comment (Perksey) [inline src/OpenAL/Extensions/OpenTK.OpenAL.Extensions.Creative.EFX/Presets/ReverbPresets.cs]: Revert all other changes to long method invocations in this file too.

File: src/OpenAL/Extensions/OpenTK.OpenAL.Extensions.Creative.EFX/Presets/ReverbProperties.cs [hunk 3]
```diff
@@ -172,8 +180,7 @@ public ReverbProperties
             float hfReference,
             float lfReference,
             float roomRolloffFactor,
-            int decayHFLimit
-        )
+            int decayHFLimit)
```
Comment (Perksey) [inline src/OpenAL/Extensions/OpenTK.OpenAL.Extensions.Creative.EFX/Presets/ReverbProperties.cs]: Revert.

File: src/OpenAL/Extensions/OpenTK.OpenAL.Extensions.EXT.Capture/AudioCapture.cs [hunk 4]
```diff
@@ -85,14 +69,12 @@ public int AvailableSamples
         /// The size of OpenAL's capture internal ring-buffer. This value expects number of samples, not
         /// bytes.
         /// </param>
-        public unsafe AudioCapture
-        (
+        public unsafe AudioCapture(
```
Comment (Perksey) [inline src/OpenAL/Extensions/OpenTK.OpenAL.Extensions.EXT.Capture/AudioCapture.cs]: Revert

File: src/OpenAL/Extensions/OpenTK.OpenAL.Extensions.EXT.Capture/AudioCapture.cs [hunk 5]
```diff
@@ -124,10 +106,8 @@ public unsafe AudioCapture
                 // Everything we tried failed. Capture may not be supported, bail out.
                 CurrentDevice = "None";
 
-                throw new AudioDeviceException
-                (
-                    "All attempts to open capture devices returned IntPtr.Zero. See debug log for verbose list."
-                );
+                throw new AudioDeviceException(
```
Comment (Perksey) [inline src/OpenAL/Extensions/OpenTK.OpenAL.Extensions.EXT.Capture/AudioCapture.cs]: Huh? Revert please

File: src/OpenAL/Extensions/OpenTK.OpenAL.Extensions.EXT.Capture/Capture.cs [hunk 6]
```diff
@@ -54,12 +61,10 @@ int size
         /// <param name="bufferFormat">The data format of the buffer.</param>
         /// <param name="sampleCount">The number of samples to retrieve.</param>
         /// <returns>The captured samples.</returns>
-        public unsafe TManagedFormat[] CaptureSamples<TManagedFormat, TBufferFormat>
-        (
+        public unsafe TManagedFormat[] CaptureSamples<TManagedFormat, TBufferFormat>(
```
Comment (Perksey) [inline src/OpenAL/Extensions/OpenTK.OpenAL.Extensions.EXT.Capture/Capture.cs]: Revert :(

File: src/OpenAL/OpenTK.OpenAL/AudioContext.cs [hunk 7]
```diff
@@ -52,6 +36,64 @@ public sealed class AudioContext : IDisposable
...
+                    if (ContextAPI.MakeContextCurrent(contextHandle))
+                    {
+                        return;
+                    }
+
+                    var contextPtr = (Context*)contextHandle.Handle;
+                    var deviceHandle = ContextAPI.GetContextsDevice(contextPtr);
+                    var error = ContextAPI.GetError(deviceHandle);
+
+                    throw new AudioContextException(
```
Comment (Perksey) [inline src/OpenAL/OpenTK.OpenAL/AudioContext.cs]: Weird method call format, revert, please.

File: src/OpenAL/OpenTK.OpenAL/AudioContext.cs [hunk 8]
```diff
@@ -145,11 +187,9 @@ public AudioContext(string device, int freq = 0, int refresh = 0, bool sync = fa
         /// For maximum compatibility, you are strongly recommended to use the default constructor.<para/>
         /// Multiple AudioContexts are not supported at this point.<para/>
         /// </remarks>
-        public AudioContext
-        (
+        public AudioContext(
```
Comment (Perksey) [inline src/OpenAL/OpenTK.OpenAL/AudioContext.cs]: Same again.

**Comment by Perksey** [comment]
I do apologize for the seemingly robotic and daunting review. A lot of this PR is predicated on the fact that the format for long method calls is wrong. It is not, ReSharper just says it is because we can't configure the way we do it. I'm pretty confident my review is just asking you to revert your changes with this regard. Speak to @Nihlus for more info regarding this.

**Comment by devvoid** [comment]
It's fine, should've asked before doing it. Fixed everything you commented on.

**Comment by Perksey** [comment]
Will run another pass over this PR, review should be in the next 3 days hopefully.

File: src/OpenAL/Extensions/OpenTK.OpenAL.Extensions.Creative.EFX/Presets/ReverbProperties.cs (hunk 3 above)
Comment (Perksey) [reply src/OpenAL/Extensions/OpenTK.OpenAL.Extensions.Creative.EFX/Presets/ReverbProperties.cs]: This has not been resolved.

File: src/OpenAL/Extensions/OpenTK.OpenAL.Extensions.EXT.Capture/Capture.cs [hunk 9]
```diff
@@ -102,11 +111,9 @@ int sampleCount
 
             if (buffer.Length < managedBufferElementCount)
             {
-                throw new ArgumentException
-                (
+                throw new ArgumentException(
```
Comment (Perksey) [inline src/OpenAL/Extensions/OpenTK.OpenAL.Extensions.EXT.Capture/Capture.cs]: Revert

File: src/OpenAL/OpenTK.OpenAL/AudioContext.cs [hunk 10]
```diff
@@ -307,22 +327,18 @@ private void CreateContext(string device, int frequency, int refreshRate, bool i
         {
             if (frequency < 0)
             {
-                throw new ArgumentOutOfRangeException
-                (
+                throw new ArgumentOutOfRangeException(
```
Comment (Perksey) [inline src/OpenAL/OpenTK.OpenAL/AudioContext.cs]: Fix please.

File: src/OpenAL/OpenTK.OpenAL/AudioContext.cs [hunk 11]
```diff
@@ -307,22 +327,18 @@ private void CreateContext(string device, int frequency, int refreshRate, bool i
...
-                    "The frequency must be greater than zero"
-                );
+                    "The frequency must be greater than zero");
             }
 
             if (refreshRate < 0)
             {
-                throw new ArgumentOutOfRangeException
-                (
+                throw new ArgumentOutOfRangeException(
```
Comment (Perksey) [inline src/OpenAL/OpenTK.OpenAL/AudioContext.cs]: And again.

File: src/OpenAL/OpenTK.OpenAL/AudioContext.cs [hunk 12]
```diff
@@ -396,11 +412,9 @@ IEnumerable<int> attributes
             if (Device == null)
             {
                 _deviceName = "None";
-                throw new AudioDeviceException
-                (
+                throw new AudioDeviceException(
```
Comment (Perksey) [inline src/OpenAL/OpenTK.OpenAL/AudioContext.cs]: Fix please

File: src/OpenAL/OpenTK.OpenAL/AudioContext.cs [hunk 13]
```diff
@@ -413,10 +427,8 @@ IEnumerable<int> attributes
             if (_contextHandle == ContextHandle.Zero)
             {
                 ContextAPI.CloseDevice(Device);
-                throw new AudioContextException
-                (
-                    "The audio context could not be created with the specified parameters."
-                );
+                throw new AudioContextException(
```
Comment (Perksey) [inline src/OpenAL/OpenTK.OpenAL/AudioContext.cs]: revert

**Comment by devvoid** [comment]
That should be everything you mentioned

**Comment by varon** [comment]
Needs some conflict resolution on your side, then I'll do a final review so we can merge.

**Pull request merged**